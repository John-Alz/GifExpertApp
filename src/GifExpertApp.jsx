import { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export default function GifExpertApp() {

    const [categories, setCategories] = useState(['Dragon ball']);

    const onAddCategory = (keyWord) => {

        if (categories.includes(keyWord)) return;

        setCategories([
            keyWord,
            ...categories
        ]);

        // setCategories(cat => [...cat, 'Valorant']) <== Importante en <AddCategory/> comentado
    };

    const onDeleteCategory = (keyWordDelete) => {
        const newArr = categories.filter(word => word !== keyWordDelete);
        setCategories(newArr)
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory={onAddCategory} />

            {
                categories.length > 0 ? categories.map((category) =>
                (
                    <GifGrid key={category} category={category} categories={categories} onDeleteCategory={onDeleteCategory} />
                )
                ) : <h2 className='empty-grid'>Busca tus gifs favoritos!!!</h2>
            }
        </>
    )
}
