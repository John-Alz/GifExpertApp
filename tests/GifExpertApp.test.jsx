import { render, screen, fireEvent } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe('Pruebas para <GifExpertApp/>', () => {

    test('Hacer match con el snapshot ', () => {

        const { container } = render(<GifExpertApp />);
        // screen.debug();

        expect(container).toMatchSnapshot();
    });

    test('Debe poderse escribir en el input', () => {

        const inputValue = 'Cars';

        render(<GifExpertApp />);

        // screen.debug();

        const input = screen.getByRole('textbox');

        fireEvent.input(input, { target: { value: inputValue } });

        expect(input.value).toBe(inputValue);
    });


    test('Debe verificar que no se agregue una categoria existente', () => {

        const category = "Dragon ball"

        render(<GifExpertApp />);

        const foundCategory = screen.getAllByText(category);

        expect(foundCategory).toHaveLength(1);
        expect(foundCategory.length).toBeLessThan(2);

    })

});
