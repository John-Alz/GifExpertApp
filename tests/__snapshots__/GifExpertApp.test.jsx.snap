// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifExpertApp/> Hacer match con el snapshot  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <div
    class="grid-container"
  >
    <div
      class="title-category-container"
    >
      <h3>
        Dragon ball
      </h3>
      <button
        aria-label="button"
      >
        <svg
          color="red"
          fill="currentColor"
          height="27"
          stroke="currentColor"
          stroke-width="0"
          style="color: red;"
          viewBox="0 0 448 512"
          width="27"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          />
        </svg>
      </button>
    </div>
    <div
      class="card-grid"
    >
      <h2>
        Cargando...
      </h2>
    </div>
  </div>
</div>
`;
